src/player.cpp

```diff
@@ -5095,9 +5093,38 @@ void player::healall(int dam)
 
 void player::hurtall(int dam, Creature *source)
 {
-    for (int i = 0; i < num_hp_parts; i++) {
-        const body_part bp = hp_to_bp( static_cast<hp_part>( i ) );
-        apply_damage( source, bp, dam );
+    if( is_dead_state() || has_trait( "DEBUG_NODMG" ) || dam <= 0 ) {
+        return;
+    }
+
+    for( int i = 0; i < num_hp_parts; i++ ) {
+        const hp_part bp = static_cast<hp_part>( i );
+        // Don't use apply_damage here or it will annoy the player with 6 queries
```

Comment:
Instead of copying most of the content of `player::apply_damage` (which leads to inconsistent behavior, `apply_damage` considers ADRENALINE trait etc.), you should move the common code into a function. That new function would be called from `hurtall` and `apply_damage`.